test/ObjectTest.cpp

```diff
@@ -1270,6 +1275,230 @@ class ObjectTest : public CppUnit::TestFixture, public ChangeListener
         CPPUNIT_ASSERT(res3->getValue() != "now");
         delete res3;
     }
+    
+    void testDateTimeObject()
+    {
+        ObjectValue* val;
+        DateTimeObject t, t2;
+        int day, month, year, wday, hour, min, sec;
+        t.setValue("2020-1-1T00:00:00");
+        CPPUNIT_ASSERT(t.getValue() == "2020-1-1T00:00:00");
+        t2.setValue("now");
+        CPPUNIT_ASSERT(t2.getValue() != "2020-1-1T00:00:00" || (sleep(2) == 0 && t2.getValue() != "2020-1-1T00:00:00"));
+
+        t.setValue("2021-5-6T17:30:05");
```

Comment:
According to [wikipedia](https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates) and an example on [ISO page](https://www.iso.org/iso-8601-date-and-time-format.html), month and day are two-digit numbers.
It's a good idea to tolerate one-digit numbers as well but let's use the standard formatting for `getValue`. DateObjectValue::toString() does not pad month and day with zeroes either but I'm fine with fixing it as well to remain consistent.